cmd/hc/main.go

```diff
@@ -65,8 +68,19 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
-	identityPath := filepath.Join(setupCfg.BasePath, "identity")
-	_, err = peertls.NewTLSFileOptions(identityPath, identityPath, true, false)
+	// GenerateCA CA
+	ca := provider.GenerateCA(context.Background(), setupCfg.CA.Difficulty, 4)
```

Comment:
should use process.Ctx(cmd) for context here. you should never need to use context.Background going forward except for in very exceptional cases

Reply:
also, this can't return an error?